src/octoprint/server/__init__.py

```diff
@@ -521,6 +522,9 @@ def on_shutdown():
 		def sigterm_handler(*args, **kwargs):
 			# will stop tornado on SIGTERM, making the program exit cleanly
 			def shutdown_tornado():
+				if self._octoprint_daemon is not None:
```

Comment:
This may be overthinking it but - don't we actually want that call to the `terminated` callback to happen after we've shutdown the ioloop instead of before? Should normally not make much of a difference I believe, but it still feels wrong to remove the pidfile while the process is still running. 

Or am I misremembering things here about the `ioloop.stop` call and its consequences and it will actually do a `sys.exit` meaning having the `terminated` call after would be too late?